Adding OneSignal Partner Source Documentation (#5499)
segmentio/segment-docs: src/connections/sources/catalog/cloud-apps/onesignal/index.md

> ## OneSignal Source

Reviewer: ```suggestion
---
title: OneSignal Source
id: o9OyD6xsVJ
beta: true
---
```

> [OneSignal](https://onesignal.com) is designed to help you send notifications and seamlessly manage your user communication across every channel, including mobile push notifications, web push notifications, in-app messaging, bulk SMS, and email. Our platform is quick to set up and makes it easy to customize and automate your messaging strategy without doing any development work.

Reviewer: ```suggestion
[OneSignal](https://onesignal.com){:target="_blank"} is designed to help you send notifications and seamlessly manage your user communication across every channel, including mobile push notifications, web push notifications, in-app messaging, bulk SMS, and email. Our platform is quick to set up and makes it easy to customize and automate your messaging strategy without doing any development work.
```

This source is maintained by Onesignal. For any issues with the source, [Contact the OneSignal Support team](mailto:support@onesignal.com)

## Getting started

> 1. From your workspace’s [Sources catalog page](https://app.segment.com/goto-my-workspace/sources/catalog) click on “Add Source”. 

Reviewer 2: ```suggestion
1. From your workspace’s [Sources catalog page](https://app.segment.com/goto-my-workspace/sources/catalog) search for OneSignal and then click on the OneSignal Source Tile. 
```

2. Search for “OneSignal” within the Sources Catalog and confirm by clicking “Add Source”
3. Give the source a nickname and a schema name. The nickname is a label used in the Segment interface, and the schema name is the namespace you query against in your warehouse. Both can be anything you like, but we recommend sticking to something that reflects the source itself and distinguishes amongst your environments (for example, OneSignal_Prod, OneSignal_Staging). Click add source to save your settings. 
4. Copy the write key from your Segment.com workspace. 
5. Log in to the OneSignal Dashboard
6. Navigate to enable the integration: Settings -> Integrations -> Segment.com and click Activate
7. Scroll down to the "Data Out" panel
8. Enter your “Write Key” from your Segment.com workspace. [Docs](https://segment.com/docs/connections/find-writekey)

> 9. Select the OneSignal Events you want to have exported, and click **Save**

Reviewer 2: ```suggestion
9. Select the OneSignal Events you want to have exported, and click **Save**. Note: Care should be taken when sending **Email Sent**,  **SMS Sent**, **Push Sent** and **In App Message Displayed** events to Segment, as doing so may contribute significantly to your Segment MTU usage volume. 
```

You are now all set to sync events from OneSignal to your Segment.com workspace. 

> From your workspace's Sources catalog page click Add Source.

Reviewer 2: Please remove

## Stream

OneSignal uses the stream Source component to send Segment event data. The events are generated from email, push notifications, in-app messages, and SMS messages that have been sent and the subsequent events data tied with it. 

> The default behavior of is for OneSignal to pass the user-level identifiers, OneSignal ID (set by OneSignal, and External ID (set by customers), with device-level identifier, Subscription ID (set by OneSignal) with each event. Events with no External ID present are anonymous events.

Reviewer: > The default behavior of is for OneSignal to pass the user-level identifiers, OneSignal ID (set by OneSignal, and External ID (set by customers), with device-level identifier, Subscription ID (set by OneSignal) with each event

Could you clarify where in the payload these will be set to `userId`, `anonymousId` or somewhere else? 

> Events with no External ID present are anonymous events.

In which case what will you use? A random anonymousId? If so please clarify

> The default behavior of is for OneSignal to pass the user-level identifiers, OneSignal ID (set by OneSignal, and External ID (set by customers), with device-level identifier, Subscription ID (set by OneSignal) with each event. Events with no External ID present are anonymous events.

Reviewer 2: ## Identifiers

The events OneSignal sends to Segment will be identifier with either userId or anonymousId. 
OneSignal will set the Segment userId equal to the OneSignal External ID. If there is no OneSignal External ID for the user, OneSignal will set the Segment anonymousId equal to the OneSignal Subscription ID. 

OneSignal does not currently send user email address or phone number details to Segment.

## Events

> The table below lists events that OneSignal sends to Segment. These events appear as tables in your warehouse, and as regular events in other Destinations. OneSignal includes the userId if available.

Reviewer 2: ```suggestion
The table below lists events that OneSignal sends to Segment. These events appear as tables in your warehouse, and as regular events in other Destinations. 
```

| Email Reported As Spam        | Email reported as Spam by recipient                           |
| Email Hardbounced             | Email returned to sender due to permanent error                                |
| Email Failed                  | Email could not deliver the email to the recipient's inbox                                     |
| SMS Sent                      | SMS sent to recipient                                        |
| SMS Delivered                 | SMS successfully Delivered                               |
| SMS Failed                    | SMS failed to send                                   |

                                  |

## Event Properties

The table below lists the properties included in the events listed above.

| Property Name     | Description                                      |
| ----------------- | ------------------------------------------------ |
| messageId         | The identifier of the discrete message           |
| campaign_id       | The identifier of the message campaign           |
| message_name      | The message name                                 |

> | message_title     | The message Title                                |

Reviewer 2: ( push and email events only)

Reviewer 3: It should also show up for SMS events. In this case, it might be more direct to state the inverse. `(Not included for IAM events)`.

Author: This is no longer applicable all fields will be relevant for each event.
